finetuner/callbacks.py

```diff
@@ -0,0 +1,138 @@
+    :param mode: one of {'auto', 'min', 'max'}. The decision to overwrite the
+        currently saved model is made based on either the maximization or the
+        minimization of the monitored quantity.
+        For an evaluation metric, this should be `max`, for `val_loss` this should
+        be `min`, etc. In `auto` mode, the mode is set to `min` if `monitor='loss'`
+        or `monitor='val_loss'` and to `max` otherwise.
+    :param verbose: Whether to log notifications when a checkpoint is saved.
+    """
+
+    save_dir: str = 'best_model_chckpt'
+    monitor: str = 'val_loss'
+    mode: str = 'auto'
+    verbose: bool = False
+
+
+@dataclass
+class TrainingCheckpoint:
+    """
+    Callback that saves the tuner state at every epoch or the last k epochs.
+
+    :param last_k_epochs: This parameter is an integer. Only the most
+        recent k checkpoints will be kept. Older checkpoints are deleted.
+    :param verbose: Whether to log notifications when a checkpoint is saved/deleted.
+    """
+
+    save_dir: str = 'training_chckpt'
+    last_k_epochs: int = 1
+    verbose: bool = False
+
+
+@dataclass
+class WandBLogger:
+    """
+    `Weights & Biases <https://wandb.ai/site>`_ logger to log metrics for training and
+    validation.
+    To use this logger, make sure to have a WandB account created, install the WandB
+    client (which you can do using ``pip install wandb``) and setting the API key as
+    environmental variable.
+
+    :param experiment: name of the experiment corresponding to the name of a
+        weights and biases project.
+    :param wandb_args: Keyword arguments that are passed to ``wandb.init`` function.
+    :param api_key: Key for wandb login.
+    """
+
+    experiment: str
+    wandb_args: dict = field(default_factory=dict)
+    api_key: Optional[str] = None
+
+
+@dataclass
+class MLFlowLogger:
+    """
+    Callback to send data to MLFlow tracking tools. The collects parameters of the
+    tuner and metrics during finetuning and validation.
+
+    For the initialization of the MLFlowLogger, the name of the experiment it
+    belongs to and a tracking_uri must be specified.
+    :param experiment: The name of the experiment of the current finetuning run.
+    :param tracking_uri: URI which refers to a storage backend. This can either be
+        a file url or a SQLAlchemy connection string. Detailed information about
+        the connection string is can be found at:
+        https://www.mlflow.org/docs/latest/python_api/mlflow.html#mlflow.set_tracking_uri
+    """
+
+    experiment: str
+    tracking_uri: str
+
+
+@dataclass()
+class LoggerCallback:
+    """
+    This callback logs training progress using a python logger.
+
+    :param batch_period: Log progress only at batches that are multiples of this
+        period.
+    :param cloudwatch: Boolean value, if sync logs to aws cloud watch.
+    """
+
+    batch_period: int = 100
+    cloudwatch: bool = False
+
+
+@dataclass
+class ProgressBarCallback:
```

Comment:
Sorry I missed that, this callback should not be exposed to the user

Reply:
makes sense